apps/datahub-e2e/src/e2e/datasets.cy.ts

```diff
@@ -456,4 +456,59 @@ describe('datasets', () => {
       })
     })
   })
+
+  describe('metadata quality', () => {
+    describe('metadata quality widget not enabled', () => {
+      it('should not show quality score sorting', () => {
+        cy.get('@sortBy').find('button').click()
+        cy.get('.cdk-overlay-container')
+          .find('[role=listbox]')
+          .find('button')
+          .should('have.length', 3)
+      })
+    })
+
+    describe('metadata quality widget enabled', () => {
+      beforeEach(() => {
+        // this will enable spatial filtering
+        cy.intercept('GET', '/assets/configuration/default.toml', {
+          fixture: 'config-with-metadata-quality.toml',
+        })
+        cy.visit('/search')
+      })
+      it('should not display widget', () => {
+        cy.get('@results')
+          .eq(2)
+          .get('gn-ui-metadata-quality')
+          .should('not.exist')
+      })
+
+      it('should reindex records', () => {
+        cy.login('admin', 'admin', false)
+        cy.visit(
+          `http://localhost:8080/geonetwork/srv/eng/admin.console#/tools`
+        )
+        cy.intercept({
+          method: 'PUT',
+          url: 'http://localhost:8080/geonetwork/srv/api/site/index?reset=false&asynchronous=true',
+        }).as('indexingRecordsXHR')
+        cy.get('[class=panel-body]').find('button').first().click()
+        cy.wait('@indexingRecordsXHR')
+      })
```

Comment:
See my comment on the docker composition; also this check here is unfortunately not enough to guarantee that all records are indeed indexed correctly, which is why I added a more robust logic in the init container. See:
https://github.com/geonetwork/geonetwork-ui/blob/ab94f4f9791994a26bea36bbfb7a4903a0357290/support-services/docker-entrypoint.d/10-index-records.sh#L50-L65

Reply:
I don't succeed to make it working that way. The `/index` curl drops index and re-index by default : 
![image](https://github.com/geonetwork/geonetwork-ui/assets/39771412/5bd5b9d1-8e75-472a-9fb9-2af000e53af0)

If I set a `reset=false` query param, it doesn't work (400 on datahub). Any idea ?

Reply:
> If I set a `reset=false` query param, it doesn't work (400 on datahub). Any idea ?

Maybe that's because the index wasn't created yet? I'm not sure. You can try using `reset=false&asynchronous=true` since is what you did in the e2e test.

Reply:
I pushed a commit trying to address that, let's see if the CI succeeds